integration-tests/src/test/groovy/com/okta/sdk/tests/it/EventHooksIT.groovy

```diff
@@ -0,0 +1,254 @@
+/*
+ * Copyright 2020-Present Okta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.okta.sdk.tests.it
+
+import com.okta.sdk.resource.ResourceException
+import com.okta.sdk.resource.EventHookChannel
+import com.okta.sdk.resource.EventHookChannelConfig
+import com.okta.sdk.resource.EventHookChannelConfigAuthScheme
+import com.okta.sdk.resource.EventHookChannelConfigHeader
+import com.okta.sdk.resource.EventSubscriptions
+import com.okta.sdk.resource.event.hook.EventHook
+import com.okta.sdk.resource.event.hook.EventHookChannelConfigAuthSchemeType
+import com.okta.sdk.resource.event.hook.EventHookList
+import com.okta.sdk.tests.it.util.ITSupport
+import org.testng.annotations.Test
+
+import static org.hamcrest.MatcherAssert.assertThat
+import static org.hamcrest.Matchers.equalTo
+import static org.hamcrest.Matchers.hasSize
+import static org.hamcrest.Matchers.notNullValue
+
+/**
+ * Tests for /api/v1/eventHooks
+ * @since 2.0.0
+ */
+class EventHooksIT extends ITSupport {
+
+    @Test
+    void createEventHookTest() {
+        String name = "event-hook-java-sdk-${UUID.randomUUID().toString()}"
+
+        EventHook createdEventHook = client.instantiate(EventHook)
+            .setName(name)
+            .setEvents(client.instantiate(EventSubscriptions)
+                .setType(EventSubscriptions.TypeEnum.EVENT_TYPE)
```

Comment:
IIRC the new open API spec should be extracting all of the ENUMs into it's own type (instead of being nested)?
cc @bretterer ?

Reply:
Not the case yet.  This is one of the next steps that we will be doing